app/controllers/admin/statistics_controller.rb

```diff
@@ -0,0 +1,22 @@
+# Admin Statistics Controller
+class Admin::StatisticsController < AdminController
+    # /admin/stats
+    def show_all
+      # query that shows all database and their joins for link_tracking
+      @databases = Database.all
+        .joins(:link_tracking)
+        .distinct
+      send_data @databases.linktracking_export, filename: "database-usage-full-#{Date.today}.csv"
+    end 
+
+    # /admin/stats/:start_date/:end_date/
+    def show
+      stats = Statistics::CSV.new({start_date: params[:start_date], end_date: params[:end_date]})
```

Comment:
Keyword params and new hash, from an already existing hash, you should rethink that and just pass params.

Reply:
updated using params.slice